core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -873,6 +873,22 @@ class GroupCoordinatorTest {
     assertEquals(Errors.FENCED_INSTANCE_ID, invalidHeartbeatResult)
   }
 
+  @Test
+  def shouldGetDifferentStaticMemberIdBasedOnTimestamp() {
```

Comment:
Does this test expose the original bug? I thought the bug would only surface if we have a loaded group metadata with old versioned format; otherwise the time stays the same but not None.

In this test `MockTimer` does not advance after the `staticMembersJoinAndRebalance` call, so the `joinTimeAsStr` would be the same?

I think we need to explicitly call `timer.advanceClock()` within the loop as well.

Reply:
We do have that within `staticMembersJoinAndRebalance`

Reply:
Yes but I'm thinking that we should also advance time within the forloop right? Otherwise the timestamp would still be static.